Patient Registry System/PatientRegistrySystem.DB/Repos/DbGenericRepository/DbGeneric.cs

```diff
@@ -0,0 +1,249 @@
+            {
+                await _entities.AddAsync(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<bool> DeleteEntityDeepAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
+                || entity.GetType().Equals(typeof(Prescription))
+                || entity.GetType().Equals(typeof(Company))
+                || entity.GetType().Equals(typeof(Medicine)))
+            {
+                _entities.Remove(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<bool> DeleteEntitySoftAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
+                || entity.GetType().Equals(typeof(Prescription))
+                || entity.GetType().Equals(typeof(Company))
+                || entity.GetType().Equals(typeof(Medicine)))
+            {
+                _entities.Update(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<List<DomainEntity>> FindAllEntitiesAsync()
+        {
+            if (typeof(DbEntity) == typeof(Record))
+            {
+                var result = _mapper.Map<List<RecordDto>>(await _applicationIdentityDbContext.Record
+                    .AsNoTracking().Where(s => (s.Status != 1)).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Prescription))
+            {
+                var result = _mapper.Map<List<PrescriptionDto>>(await _applicationIdentityDbContext.Prescription
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Company))
+            {
+                var result = _mapper.Map<List<CompanyDto>>(await _applicationIdentityDbContext.Company
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Medicine))
+            {
+                var result = _mapper.Map<List<MedicineDto>>(await _applicationIdentityDbContext.Medicine
+                    .AsNoTracking().ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public async Task<List<DomainEntity>> FindAllEntitiesAsync(int entityId)
+        {
+            if (typeof(DbEntity) == typeof(Record))
```

Comment:
same comment as above, and we need a more generic way to filter by entityId. You can get the PK column for any table and then compare to entityId.